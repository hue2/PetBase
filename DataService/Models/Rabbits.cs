using Petbase.DataService.Interfaces;

namespace Petbase.DataService.Models
{
    public partial class Rabbits : IEntity
    {
        public string Id { get; set; }
        public string Name { get; set; }
        public string LifeSpan { get; set; }
        public string UniqueTraits { get; set; }
        public string Size { get; set; }
        public string Coat { get; set; }
        public string PictureUrl { get; set; }
    }
}
