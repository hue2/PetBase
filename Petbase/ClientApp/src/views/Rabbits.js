import React from 'react';
import Table from './Table/Table';
import RabbitApi from '../api/RabbitApi';
import Modal from './Modal/Modal';
import Columns from './Table/Columns';

export default class Rabbits extends React.PureComponent {
    state = {
        data: [],
        showModal: false,
        selectedRow: { name: '', imageUrl: '' },
    }

    handleImageClick = (row) => {
        this.setState({ selectedRow: row });
        this.toggleModal();
    }

    rabbitApi = new RabbitApi();
    columns = new Columns().getColumns(this.handleImageClick);

    componentDidMount () {
        this.rabbitApi.get().then(data => {
            this.setState({ data });
        })
    }

    toggleModal = () => {
        this.setState({
            showModal: !this.state.showModal,
        });
    }

    render() {      
        return (
            <div>
                <Table data={this.state.data} columns={this.columns} title={"Rabbits"}/>
                <Modal 
                    showModal={this.state.showModal} 
                    title={this.state.selectedRow.name} 
                    toggle={this.toggleModal} 
                    imageUrl={this.state.selectedRow.pictureUrl} 
                />
            </div>
        )
    }
}