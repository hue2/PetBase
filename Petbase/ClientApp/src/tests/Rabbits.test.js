import React from 'react';
import {shallow} from 'enzyme/build';
import Rabbits from '../views/Rabbits';
import renderer from 'react-test-renderer';

it('mounts without crashing', () => {
  const wrapper = shallow(<Rabbits />);
  wrapper.unmount()
});

it('renders correctly', () => {
    const tree = renderer
      .create(<Rabbits />)
      .toJSON();
    expect(tree).toMatchSnapshot();
});