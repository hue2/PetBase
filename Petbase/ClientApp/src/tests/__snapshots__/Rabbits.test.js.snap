// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="animated fadeIn"
  >
    <div
      className="row"
    >
      <div
        className="col"
      >
        <br />
        <div>
          <h2>
            Rabbits
          </h2>
          <hr />
        </div>
        <div
          className="card"
        >
          <div
            className="card-body"
          >
            <div>
              <label
                className="search-label"
                htmlFor="search-bar-0"
              >
                <span
                  className="sr-only"
                >
                  Search this table
                </span>
                <input
                  aria-label="enter text you want to search"
                  className="form-control "
                  id="search-bar-0"
                  onChange={[Function]}
                  onKeyUp={[Function]}
                  placeholder="Search"
                  style={
                    Object {
                      "marginBottom": 20,
                    }
                  }
                  type="text"
                  value=""
                />
              </label>
              <div
                className="react-bootstrap-table"
              >
                <table
                  className="table table-striped table-hover"
                >
                  <thead>
                    <tr>
                      <th
                        style={
                          Object {
                            "width": "15%",
                          }
                        }
                        tabIndex={0}
                      >
                        Name
                      </th>
                      <th
                        style={
                          Object {
                            "width": "15%",
                          }
                        }
                        tabIndex={0}
                      >
                        Life Span
                      </th>
                      <th
                        style={
                          Object {
                            "width": "15%",
                          }
                        }
                        tabIndex={0}
                      >
                        Size
                      </th>
                      <th
                        style={
                          Object {
                            "width": "15%",
                          }
                        }
                        tabIndex={0}
                      >
                        Coat
                      </th>
                      <th
                        style={
                          Object {
                            "width": "25%",
                          }
                        }
                        tabIndex={0}
                      >
                        Unique Traits
                      </th>
                      <th
                        tabIndex={0}
                      >
                        Picture
                      </th>
                    </tr>
                  </thead>
                </table>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
